client/X11/xf_window.c

```diff
@@ -185,30 +185,27 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 		 */
 		startX = startX + xfc->instance->settings->MonitorLocalShiftX;
 		startY = startY + xfc->instance->settings->MonitorLocalShiftY;
-	}
-
-	xf_ResizeDesktopWindow(xfc, window, width, height);
 
-	/* Set the fullscreen state */
-	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
-				fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
-				xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
-
-	/* Only send monitor bounds if they are valid */
-	if ((xfc->fullscreenMonitors.top >= 0) &&
-			(xfc->fullscreenMonitors.bottom >= 0) &&
-			(xfc->fullscreenMonitors.left >= 0) &&
-			(xfc->fullscreenMonitors.right >= 0))
-	{
-		xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_FULLSCREEN_MONITORS, 5,
+		/* Set monitor bounds */
+		if (settings->MonitorCount > 1)
```

Comment:
@bjcollins ... and with this change I got at least single monitor fullscreen (/f) working and displaying on the correct monitor (either with your mouse position detection and/or by specifying a single monitor id using /f /monitors:X) with all window managers I had available (xfce, unity, openbox 3.5/3.6, icewm, awesome)